src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck.java

```diff
@@ -464,12 +464,12 @@ private void logCommaViolation(final DetailAST ast) {
         //comma can be null if array is empty
         final DetailAST comma = rCurly.getPreviousSibling();
 
-        if (trailingArrayComma == TrailingArrayComma.ALWAYS
-            && (comma == null || comma.getType() != TokenTypes.COMMA)) {
-            log(rCurly, MSG_KEY_ANNOTATION_TRAILING_COMMA_MISSING);
+        if (trailingArrayComma == TrailingArrayComma.ALWAYS) {
+            if (comma == null || comma.getType() != TokenTypes.COMMA) {
+                log(rCurly, MSG_KEY_ANNOTATION_TRAILING_COMMA_MISSING);
+            }
         }
-        else if (trailingArrayComma == TrailingArrayComma.NEVER
-            && comma != null && comma.getType() == TokenTypes.COMMA) {
+        else if (comma != null && comma.getType() == TokenTypes.COMMA) {
```

Comment:
`trailingArrayComma == TrailingArrayComma.NEVER` was removed because we are already check that the field can only be 2 of 3 possible values since it is an enumeration. Since we have the check outside this method, this extra was unnecssary and hence why it was removed.